packages/cozy-sharing/src/components/Recipient.spec.jsx

```diff
@@ -95,6 +96,94 @@ describe('Recipient component', () => {
     expect(onRevoke).toBeCalled()
     expect(onRevokeSelf).not.toBeCalled()
   })
+
+  it('should render confirmation actions if recipient is waiting for confirmation', async () => {
+    const confirmRecipient = jest.fn()
+    const rejectRecipient = jest.fn()
+
+    const { getByText } = setup({
+      instance: 'foo.mycozy.cloud',
+      status: 'ready',
+      type: 'two-way',
+      isOwner: false,
+      documentType: 'Organizations',
+      recipientConfirmationData: {
+        name: 'Bob'
+      },
+      rejectRecipient,
+      confirmRecipient
+    })
+
+    expect(getByText('Confirm')).toBeTruthy()
+  })
+
+  it('should not render confirmation actions if no recipient is waiting for confirmation', async () => {
+    const confirmRecipient = jest.fn()
+    const rejectRecipient = jest.fn()
+
+    const { queryByText } = setup({
+      instance: 'foo.mycozy.cloud',
+      status: 'ready',
+      type: 'two-way',
+      isOwner: false,
+      documentType: 'Organizations',
+      recipientConfirmationData: undefined,
+      rejectRecipient,
+      confirmRecipient
+    })
+
+    expect(queryByText('Confirm')).not.toBeInTheDocument()
+  })
+
+  it(`should call confirmRecipient when clicking 'confirm' button`, async () => {
+    const confirmRecipient = jest.fn()
+    const rejectRecipient = jest.fn()
+
+    const { getByText } = setup({
+      instance: 'foo.mycozy.cloud',
+      status: 'ready',
+      type: 'two-way',
+      isOwner: false,
+      documentType: 'Organizations',
+      recipientConfirmationData: {
+        name: 'Bob'
+      },
```

Comment:
Do you mean using the same instance of `recipitientConfirmationData` in both `setup()` and `toBeCalledWith()` calls?
What if the component has a side effect on recipientConfirmationData? The expect would still pass and I think we would missed a bug detection here. 

> Like that if we add informations

> "Are we really confirming by using the name of the user and not its id?"

I don't think this would change in the future in the context of this test.
This object is intentionally declared as `PropTypes.object` because we don't manipulate it in `Recipient`, it is only passed as an argument of `confirmRecipient(recipientConfirmationData)`.
Therefore we don't bother what is inside this object, the only requirement is that it is correctly sent to `confirmRecipient()`.
And that's why the test only declares a name, it is the minimal content to check that the same object is passed as argument.

Do you still think I should use an object reference for the test?

In fact there is another requirement as it should have an `email` member when being manipulated by `WhoAsAccess` (the component that instantiate `Recipient`).

Maybe I should add this proptype to `WhoAsAccess`
```js
recipientsToBeConfirmed: React.PropTypes.arrayOf(React.PropTypes.shape({
     email: React.PropTypes.string.isRequired
}))
```

Also I can change `name: 'Bob'` for `email: 'xxx'` in order to have a test that act more like a documentation.

Reply:
> Also I can change name: 'Bob' for email: 'xxx' in order to have a test that act more like a documentation.

It's alway a good thing to have tests acting like documentation and working with the same data as in production.

Reply:
Fixed, tests are now using email and WhoAsAccess has the correct PropType